tests/unit/amo/components/TestAddon.js

```diff
@@ -91,11 +118,123 @@ describe('Addon', () => {
     sinon.assert.calledWith(fakeDispatch, setViewContext(fakeAddon.type));
   });
 
+  it('dispatches setViewContext on update', () => {
+    const fakeDispatch = sinon.stub();
+    const root = render({ dispatch: fakeDispatch });
+    fakeDispatch.reset();
+    root.componentWillReceiveProps({
```

Comment:
Cool, let me know if you think it warrants another review; should be easy to do and quick!